tests/src/whisk/core/WskDeadlineLongTests.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import common.TestHelpers
+import common.TestUtils
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol._
+import spray.json._
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+@RunWith(classOf[JUnitRunner])
+class WhiskDeadlineLongTests
+   extends TestHelpers
+    with WskTestHelpers {
+
+    behavior of "Whisk deadline long running"
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk
+    val defaultAction = Some(TestUtils.getTestActionFilename("hello.js"))
+
+    it should s"invoke a nodejs action that returns a result by the deadline of 5 minutes" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "deadline"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("helloDeadline.js")), timeout = Some(300 seconds))
+            }
+
+            val run = wsk.action.invoke(name)
+            withActivation(wsk.activation, run, totalWait = 310 seconds) {
```

Comment:
the default poll frequency is too high for a test that's expected to last this long.